lib/BloqadeLattices/src/interact.jl

```diff
@@ -18,9 +18,9 @@ julia> two_body_interaction_matrix(atoms) do x,y return 1/distance(x,y) end
 ```
 """
 function two_body_interaction_matrix(f, atoms)
-
     natoms = length(atoms)
-    mat = zeros(natoms,natoms)
+    T = (atoms isa AbstractVector && !isempty(atoms)) ? typeof(atoms[1][1]) : Float64
```

Comment:
add a test?

Reply:
Should I add tests in BloqadeKrylov and BloqadeODE?

Reply:
BloqadeKrylov